src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php

```diff
@@ -93,6 +104,12 @@ private function createNewPayment(BaseOrderInterface $order)
         }
 
         $payment->setMethod($paymentMethod);
+
+        if (OrderCheckoutStates::STATE_COMPLETED === $order->getCheckoutState()) {
+            $stateMachine = $this->stateMachineFactory->get($payment, 'sylius_payment');
```

Comment:
This should not be done here, there is already a cascade transition on "complete" transition IIRC, can you check @Zales0123?